CodeEdit/Features/AppPreferences/Sections/TextEditingPreferences/TextEditingPreferencesView.swift

```diff
@@ -130,6 +133,72 @@ struct TextEditingPreferencesView: View {
         }
     }
 
+    private var textDecoding: some View {
+        Picker("Text decoding", selection: $prefs.preferences.textEditing.textDecoding) {
+            Group {
+                Text("ASCII")
+                    .tag(AppPreferences.TextDecodingFormats.ascii)
+                Text("Non Lossy ASCII")
+                    .tag(AppPreferences.TextDecodingFormats.nonLossyASCII)
+
+                Divider()
+
+                Text("ISO 2022 Japan")
+                    .tag(AppPreferences.TextDecodingFormats.iso2022JP)
+                Text("Japanese EUC")
+                    .tag(AppPreferences.TextDecodingFormats.japaneseEUC)
+
+                Divider()
+
+                Text("ISO Latin 1")
+                    .tag(AppPreferences.TextDecodingFormats.isoLatin1)
+                Text("ISO Latin 2")
+                    .tag(AppPreferences.TextDecodingFormats.isoLatin2)
+                Text("macOS Roman")
+                    .tag(AppPreferences.TextDecodingFormats.macOSRoman)
+            }
+
+            Group {
+                Divider()
+
+                Text("Nextstep")
+                    .tag(AppPreferences.TextDecodingFormats.nextstep)
+                Text("Shift JIS")
+                    .tag(AppPreferences.TextDecodingFormats.shiftJIS)
+                Text("Symbol")
+                    .tag(AppPreferences.TextDecodingFormats.symbol)
+
+                Divider()
+            }
+
+            Group {
+                Text("Unicode")
+                    .tag(AppPreferences.TextDecodingFormats.unicode)
+
+                Text("Unicode (UTF-8)")
+                    .tag(AppPreferences.TextDecodingFormats.utf8)
+
+                Divider()
+
+                Text("Unicode (UTF-16)")
+                    .tag(AppPreferences.TextDecodingFormats.utf16)
+                Text("Unicode (UTF-16) Big Endian")
+                    .tag(AppPreferences.TextDecodingFormats.utf16be)
+                Text("Unicode (UTF-16) Little Endian")
+                    .tag(AppPreferences.TextDecodingFormats.utf16le)
+
+                Divider()
+
+                Text("Unicode (UTF-32)")
+                    .tag(AppPreferences.TextDecodingFormats.utf32)
+                Text("Unicode (UTF-32) Big Endian")
+                    .tag(AppPreferences.TextDecodingFormats.utf32be)
+                Text("Unicode (UTF-32) Little Endian")
+                    .tag(AppPreferences.TextDecodingFormats.utf32le)
+            }
+        }
+    }
```

Comment:
When using `String.Encoding` you could add an extension like so:

```swift
extension String.Encoding {
    static var all: [String.Encoding] {
        [
            .ascii,
            .iso2022JP,
            .isoLatin1,
            ...
        ]
    }
```

Which you can then loop over inside the picker. For the display string just get the `description` of the Encoding. Set the Tag to the Encoding's `rawValue`

A grouping like in Xcode would also be preferable:

<img width="304" alt="Screenshot 2023-03-25 at 10 32 58" src="https://user-images.githubusercontent.com/9460130/227709166-1e037307-f062-4c9b-bd2c-d626afe6f93e.png">

Reply:
The grouping could be accomplished by adding further extensions to `String.Encoding` like the `all` property mentioned above.

```swift
static var allUnicode: [String.Encoding] {...}
static var allMacOS: [String.Encoding] {...}
static var allISO: [String.Encoding] {...}
static var allWindows: [String.Encoding] {...}
static var allDOS: [String.Encoding] {...}
static var allEUC: [String.Encoding] {...}
static var others: [String.Encoding] {...}
```